sktime/libs/timemoe/ts_generation_mixin.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2023, Salesforce, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ts_generation_mixin."""
+
+import warnings
+from typing import Any, Optional, Union
+
+from sktime.utils.dependencies import _check_soft_dependencies
+
+if _check_soft_dependencies("torch", severity="none"):
```

Comment:
the dependencies could all be moved inside the class - that is more robust against changes in the soft dependencies.

For the `config`, you could have a private method `_get_default_config` that returns the current dict.